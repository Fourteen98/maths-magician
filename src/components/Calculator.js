import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  render() {
    return (
      <div className="container">
        <div className="wrapper">
          <div className="screen"> 0</div>
          <button type="button" className="btn num-pad">AC</button>
          <button type="button" className="btn num-pad">+ / -</button>
          <button type="button" className="btn num-pad">%</button>
          <button type="button" className="btn orange">รท</button>
          <button type="button" className="btn num-pad">7</button>
          <button type="button" className="btn num-pad">8</button>
          <button type="button" className="btn num-pad">9</button>
          <button type="button" className="btn orange">x</button>
          <button type="button" className="btn num-pad">4</button>
          <button type="button" className="btn num-pad">5</button>
          <button type="button" className="btn num-pad">6</button>
          <button type="button" className="btn orange">-</button>
          <button type="button" className="btn num-pad">1</button>
          <button type="button" className="btn num-pad">2</button>
          <button type="button" className="btn num-pad">3</button>
          <button type="button" className="btn orange">+</button>
          <button type="button" className="btn num-pad num-zero">0</button>
          <button type="button" className="btn num-pad">.</button>
          <button type="button" className="btn orange">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
