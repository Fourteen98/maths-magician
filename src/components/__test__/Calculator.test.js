import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('renders when there is item', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
