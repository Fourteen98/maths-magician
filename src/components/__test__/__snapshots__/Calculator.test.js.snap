// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is item 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="screen"
    />
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      + / -
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn num-pad num-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn num-pad"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
